client/catapult/vcpkg.json

```diff
@@ -0,0 +1,678 @@
+{
+  "builtin-baseline": "b346ee32f13f3bf2ef19ac02f982f6938f7a84ea",
+  "dependencies": [
+    "benchmark",
+    "boost-atomic",
+    "boost-chrono",
+    "boost-date-time",
+    "boost-dll",
+    "boost-filesystem",
+    "boost-log",
+    "boost-math",
+    "boost-multiprecision",
+    "boost-program-options",
+    "boost-regex",
+    "boost-thread",
+    "boost-regex",
+    "boost-system",
+    "cppzmq",
+    "gtest",
+    "mongo-cxx-driver",
+    "openssl",
+    "rocksdb"
+  ],
+  "overrides": [
+    {
+      "name": "benchmark",
+      "version": "1.8.3#3"
+    },
+    {
+      "name": "boost",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-accumulators",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-algorithm",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-align",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-any",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-array",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-asio",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-assert",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-assign",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-atomic",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-beast",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-bimap",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-bind",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-build",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-callable-traits",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-chrono",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-circular-buffer",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-cobalt",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-compat",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-compatibility",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-compute",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-concept-check",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-config",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-container",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-container-hash",
+      "version": "1.83.0#0"
+    },
+    {
+      "name": "boost-context",
```

Comment:
Catapult does not need all the boost libs like ``context, contract, etc``.
Here is a list of what is removed - https://github.com/symbol/symbol/blob/b3c472b6b36eaf31e5f1d939ceb5b9b7e0f1186d/jenkins/catapult/baseImageDockerfileGenerator.py#L11

Reply:
A bit of explanation is needed here.
The list of Catapult dependencies is in the section "dependencies" of that json file.
What you find in the section "overrides" has multipurpose:

1. The way vcpkg segments the Boost library is a bit different and more granular than "normal" boost libs
2. Many of the boost components require ancillary components which are not explicitly indicated in catapult dependencies
3. To ensure all components  of the boost set are built on the same version of the library I simply grabbed ALL possible boost targets built by vcpkg enforcing on them (in overrides section) the specific version 1.83

For instance this is the list of Boost's components built as dependencies of the Catapult's boost required components
```
1> [CMake]   * boost-algorithm:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-algorithm\064c9bd0deddb764f57008b07c0d42af035c1a50
1> [CMake]   * boost-align:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-align\6312ca6cedb77e48b123aa19e7ef98f5401b4020
1> [CMake]   * boost-any:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-any\8995a49c0eb70b1e306323f70d79dc5b6343bd6b
1> [CMake]   * boost-array:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-array\98d20d502bbd77d5a4312254bd47d83c55b4bbac
1> [CMake]   * boost-asio:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-asio\0d4a6f6c528811c3c79ac046e738e5bab3db995f
1> [CMake]   * boost-assert:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-assert\058c55df927f8a8180003c576bb4b536a46cf3e8
1> [CMake]   * boost-bind:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-bind\b846d10f5df5d2d6f27bd2fe13170cf3c09d8cbf
1> [CMake]   * boost-build:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-build\bd9d972c221f9e5bdd839246960791cdcfe80524
1> [CMake]   * boost-concept-check:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-concept-check\ec7a65ff7072399f9a926316cac0b1c6162d4ee5
1> [CMake]   * boost-config:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-config\26e9082aeb3829b11d9f0ea82de37db9a93b6620
1> [CMake]   * boost-container:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-container\06dc49b47f365371fd98d0373e46369bb97056c8
1> [CMake]   * boost-container-hash:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-container-hash\058bb680534a2375ec78dd366d9969967bd7570e
1> [CMake]   * boost-context:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-context\2c0ccde97a196bd6c78f0fbb52ae9c8b33a525b5
1> [CMake]   * boost-conversion:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-conversion\45d1d99389896f68454518ac9e7d516c88598fcd
1> [CMake]   * boost-core:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-core\736b9319578faba560f1bccd62ae53421c290ac4
1> [CMake]   * boost-coroutine:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-coroutine\d8a5c2cff3de497580fc8dc363ebc3e8ec0c9a1d
1> [CMake]   * boost-describe:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-describe\5287728efcb90f341cbdbf3e40edd2568baa3d98
1> [CMake]   * boost-detail:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-detail\9a596ae27e05912ce6ac9430ad0a2c60b0874517
1> [CMake]   * boost-dynamic-bitset:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-dynamic-bitset\783e8e0c0387114a445ad62dacbbb2f2ce910cd2
1> [CMake]   * boost-endian:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-endian\5321b25b38e55ebe1bbeb54242c104e208b643f8
1> [CMake]   * boost-exception:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-exception\8dce130433b2a5fc44ee576f9b39bb97d916661e
1> [CMake]   * boost-format:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-format\8b4697f0c1e3cc49f4bf08c12b748544c132db48
1> [CMake]   * boost-function:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-function\f066c30c0481a6f75452d3a1372ea3425adea156
1> [CMake]   * boost-function-types:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-function-types\77f27d230b3784277075862bba2c282977ea1f0a
1> [CMake]   * boost-functional:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-functional\aba27201e0794309bf68a44e8727378345927b38
1> [CMake]   * boost-fusion:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-fusion\0475f7220a08b37bfbea03b873211f9ea76fac08
1> [CMake]   * boost-integer:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-integer\f70c3d065ad54b0b64e16541cfdd86aceec513f3
1> [CMake]   * boost-interprocess:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-interprocess\685634ef2e0b7c514fc454d668e27c17937dc005
1> [CMake]   * boost-intrusive:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-intrusive\05886d36db2ae6ecb44cf3b964c2d0a4d486c129
1> [CMake]   * boost-io:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-io\f5d84659eada088f612945a42ae940ee7dcc37be
1> [CMake]   * boost-iterator:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-iterator\8870bee36b1dee543b554c3e0264e6871a6ad6e2
1> [CMake]   * boost-lexical-cast:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-lexical-cast\dde0974bedf5d7539835eeadb3183b72d0df75f0
1> [CMake]   * boost-modular-build-helper:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-modular-build-helper\843c4331dc1453c80f99e25be4513c99074c5cf2
1> [CMake]   * boost-move:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-move\3f4770bf9904c733be328e32d4017a0ed1157d2d
1> [CMake]   * boost-mp11:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-mp11\68c2556b43afbb853b5e724eb0fd5269a92088f1
1> [CMake]   * boost-mpl:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-mpl\dfcb80d8d1b767426e6714d6ba1b6a1473c0cc23
1> [CMake]   * boost-multi-index:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-multi-index\7d40ee58c3e54b9c841abc6dd57033366218f064
1> [CMake]   * boost-numeric-conversion:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-numeric-conversion\c3a65f3521df33b8a76605097a3046795f1107e7
1> [CMake]   * boost-optional:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-optional\145809a6e99a8ba579c3c0bef2c48969555a2bd5
1> [CMake]   * boost-parameter:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-parameter\1dce1428072cad1f8bc7b6c02f14cc6ba12da3fd
1> [CMake]   * boost-phoenix:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-phoenix\a4c5832e8ebf68cd4f2ee601a10aa7adfc12e6e9
1> [CMake]   * boost-pool:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-pool\e5c81fe62cb0da4fdc940d21d87837678a2e67cf
1> [CMake]   * boost-predef:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-predef\8e76add266584a1177e9a0f2c59c43a78d88e578
1> [CMake]   * boost-preprocessor:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-preprocessor\07609446e5cce5a0d9afed4570e0664708444b54
1> [CMake]   * boost-property-tree:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-property-tree\caa9ef5866c4dcf2869fca47fb3a478f74e07f7a
1> [CMake]   * boost-proto:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-proto\7bb3591b035347ce419b85ae321c0e1b6c71084a
1> [CMake]   * boost-random:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-random\045960cb6d4295d363e1748931ea56fc245f7c09
1> [CMake]   * boost-range:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-range\7d5d869d86b0abdd19022ee0b0c5f24192769291
1> [CMake]   * boost-ratio:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-ratio\0fafb812427bea7d49dd318f3e72778149f443ab
1> [CMake]   * boost-rational:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-rational\55ec23b1953b2e44dc43fda4547cf0b96bfe140f
1> [CMake]   * boost-serialization:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-serialization\7ebe53455e9721a76ebe14c651a2deafbcb758b8
1> [CMake]   * boost-smart-ptr:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-smart-ptr\81506ee1238bf93218cef84de86dca96dd0b3682
1> [CMake]   * boost-spirit:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-spirit\e2ebc2bcf72d8ae953ac54163a7c5b43f17396f0
1> [CMake]   * boost-static-assert:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-static-assert\50eac53fdc886f9de87fc06717a4ae3a939b28e9
1> [CMake]   * boost-throw-exception:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-throw-exception\22bf09f8d91024cc1e74b19bb6333baeea262b69
1> [CMake]   * boost-tokenizer:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-tokenizer\05ada2a1cc63fe9b4f5b4e9ac1d6ec63961e281b
1> [CMake]   * boost-tuple:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-tuple\a5a7d3cf0427278418ddc0e91a3858218535c8a9
1> [CMake]   * boost-type-index:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-type-index\58d41ff7ca046c8997ddd3f6c8ab801291edd9c4
1> [CMake]   * boost-type-traits:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-type-traits\6595fdcc50c1d1d5587ade48484a0b074ddb079b
1> [CMake]   * boost-typeof:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-typeof\ec85052de49f1335b474cee0689971b9ad4df92c
1> [CMake]   * boost-uninstall:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-uninstall\1dfbff4a218e77e371ebc397f0573859136c7c92
1> [CMake]   * boost-unordered:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-unordered\d78bed13cd960d7723ac5b0c82d2223c321ee0a9
1> [CMake]   * boost-utility:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-utility\175e91f86d24ff49330e7f5f95db441ceb85886e
1> [CMake]   * boost-variant:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-variant\60ca10585ea590b82692a0f715cfcc596d4799db
1> [CMake]   * boost-variant2:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-variant2\a249f8469b4c3ccb2aee3ece628d69a71d54b35f
1> [CMake]   * boost-vcpkg-helpers:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-vcpkg-helpers\18a50fedd4344db5ce1f82546be1889215381f78
1> [CMake]   * boost-winapi:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-winapi\cfd68f543033ed6b8cee9235ef287b7fd4f8709a
1> [CMake]   * boost-xpressive:x64-windows@1.83.0 -- E:\vcpkg\buildtrees\versioning_\versions\boost-xpressive\55de0dfb037123cd093c56acf61b22948e98295b
```

If it wasn't so picture this: I only specify we need boost's `chrono` at version 1.83 but it also requires (as internal dependency) say, `boost-numeric-conversion`. As version for a component is not propagated by vcpkg to other deps it would happen we get `chrono` at version 1.83 but linked to `numeric-conversion` at version 1.84 (which is boost's latest).

Reply:
Thanks for the explanation. It is weird to link the dependency module from another version.

Reply:
This better shows what I'm talking about
https://vcpkg.link/ports/boost-atomic/v/1.83.0/0

If not specified in the override Boost's assert would have been built on version 1.85 (latest)